src/profile-logic/sanitize.js

```diff
@@ -362,6 +374,125 @@ function sanitizeThreadPII(
     delete newThread['eTLD+1'];
   }
 
+  if (
+    PIIToBeRemoved.shouldRemovePrivateBrowsingData &&
+    windowIdToBeSanitized.size > 0
+  ) {
+    // In this block, we'll remove everything related to frame table entries
```

Comment:
This block that does the frame sanitization looks good to me, thanks for doing it!